src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -66,7 +66,7 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('collectors.xml');
         $loader->load('guard.xml');
 
-        if ($container->hasParameter('kernel.debug') && $container->getParameter('kernel.debug')) {
+        if ($container->has('profiler')) {
```

Comment:
It's about this:
```php
if ($container->has('profiler')) {
    // ...
}
```
Versus this:
```php
if ($container->getParameter('kernel.debug')) {
    // ...
}
```

The `$container` parameter of `Extension::load()` is *not* the actual container, it's a copy. See https://github.com/symfony/dependency-injection/blob/master/Compiler/MergeExtensionConfigurationPass.php#L47-L49 Only the parameters are copied from actual to tmp container, services aren't.

This is why `$container->has('profiler')` will *always* return false.